src/riak_kv_qry_worker.erl

```diff
@@ -258,6 +258,28 @@ throttling_spawn_index_fsms(State) ->
     State.
 
 
+estimate_query_size(#state{n_subqueries_done = NSubqueriesDone,
+                           max_query_data    = MaxQueryData,
+                           sub_qrys          = SubQrys,
+                           result            = AggregateResult,
+                           qry = ?SQL_SELECT{'SELECT' = #riak_sel_clause_v1{calc_type = CalcType}} =
+                               OrigQry} = State)
+  when CalcType /= rows,
+       NSubqueriesDone == 1 ->
```

Comment:
For `aggregate` queries, the structure in `#state.result` does not grow in size. It is set up initially, at query compilation, with initial states of all groups (columns like `AVG(x)`) which `riak_kv_qry_worker:add_subquery_result`, eventually calling `riak_kv_qry_compiler:run_select`, updates with every collected row.

For `group_by` queries, the result set _does_ grow. A query such as `AVG(x) GROUP BY y` will grow to the number of distinct values in column `y`, which in extreme cases can represent a situation where OOM protection is very much applicable.

d53e0d0 is a properer version of the size checks.